plugins/autoslab.cpp

```diff
@@ -0,0 +1,267 @@
+/* Simple plugin to check for ghosts and automatically queue jobs to engrave slabs for them.
+ *
+ * Enhancement idea: Queue up a ConstructSlab job, then link the engrave slab job to it. Avoids need to have slabs in stockpiles
+ *                   Would require argument parsing, specifying materials
+ * Enhancement idea: Automatically place the slab. This seems like a tricky problem but maybe solveable with named zones?
+ *                   Might be made obsolete by people just using buildingplan to pre-place plans for slab?
+ * Enhancement idea: Optionally enable autoengraving for pets.
+ * Enhancement idea: Try to get ahead of ghosts by autoengraving for dead dwarves with no remains.
```

Comment:
or unreachable corpses (e.g. at the bottom of a pool or in a closed-off cavern -- see autochop.cpp for how I detect reachability for logs)

Reply:
Good idea, I'll append the enhancement comment.